Forward fix the missing torch.nn.Module.set_submodule from D59140215

huydhn opened the pull request:
Summary: This is to forward fix D59140215 from a PyTorch open source contributor T194074371. On PyTorch side, we need to use isinstance instead of type when checking for nn.Module.  This is the same way get_submodule is currently implemented.

Test Plan: `buck2 test 'fbcode//mode/opt' fbcode//dper3/dper3/core/tests:module_test`

Differential Revision: D59254638

mikaylagawarecki reviewed (approved):
Thanks, and my bad for not catching this when reviewing, there was actually a fix in https://github.com/pytorch/pytorch/pull/129837 but let's merge this one instead

huydhn commented:
> Thanks, and my bad for not catching this when reviewing, there was actually a fix in #129837 but let's merge this one instead

Oh, I see.  Thank you for the review.  Let me check if the change from the test case of https://github.com/pytorch/pytorch/pull/129837 is needed, so I can copy it over here.

huydhn committed:
Forward fix the missing torch.nn.Module.set_submodule from D59140215 (#130075)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/130075

This is to forward fix D59140215 from a PyTorch open source contributor T194074371.  On PyTorch side, we need to use `isinstance` instead of `type` when checking for `nn.Module`.  This is the same way `get_submodule` is currently implemented.

Test Plan: `buck2 test 'fbcode//mode/opt' fbcode//dper3/dper3/core/tests:module_test`

Reviewed By: mikaylagawarecki

Differential Revision: D59254638
